test/transformer.spec.ts

```diff
@@ -1453,6 +1453,45 @@ describe('transformer', () => {
             expect(ageHidden.hasAttribute('data-setgeopoint')).to.equal(true);
         });
     });
+
+    describe('Optimization regression tests', () => {
+        // These tests are meant to exercise the second condition for definding
+        // `data-form-id`. The XSL seems to suggest that when an explicit `id`
+        // is not present, the value of `xmlns` will be copied instead. That
+        // behavior would be quite odd! Nonetheless, these tests ensure that the
+        // behavior doeesn't change.
```

Comment:
As far as I know, this exists to try to make Enketo usable by CommCare. CommCare and ODK share a lot of history but quickly deviated from each other. CommCare uses `xmlns` in some [unexpected ways](https://github.com/XLSForm/pyxform/issues/105#issuecomment-1305977323) including as a way to uniquely identify a form definition. 

As far as I know this is not used in Enketo.

Reply:
My best guess is (also) that this was found in one of the forms originally hosted on ODK's aggregate.appspot.com (i.e. a form without an `id` but with a namespace. Similarly, that's the only place I found forms with relative nodeset (and maybe ref) attributes, I believe (eIMCI form or something), which also added a lot of slowdown and code though that was later commented out.

Great to remove this!

Reply:
🎉